src/utils/ParseRouteUtils.js

```diff
@@ -532,8 +535,135 @@ function parseRoute(resp: Object, destinationName: string) {
+    const paths = await parseRoute(resp, destinationName);
+    
+    return paths.map((currPath) => {
+        const {
+            arrivalTime,
+            departureTime,
+            directions,
+            endCoords,
+            startCoords,
+        } = currPath;
+
+        let travelDistance = geolib.getDistance(
+            { latitude: startCoords.lat, longitude: startCoords.long },
+            { latitude: endCoords.lat, longitude: endCoords.long }
+        );
+
+        travelDistance *= METERS_TO_MILES;
+        currPath.travelDistance = travelDistance;
+
+        let totalDuration = new Date(arrivalTime) - new Date(departureTime);
+        totalDuration /= ONE_MINUTE_IN_MILLISECONDS;
+        currPath.totalDuration = Math.round(totalDuration);
+
+        currPath.startName = currPath.directions[0].name;
+        currPath.endName = destinationName;
+
+        // routeSummary
+
+        /* routeSummaryElement {
+            stopName: String,
+            direction: RouteSummaryDirection?,
+            shouldStayOnBus: Bool,
+        }
+        */
+
+        /* routeSummaryDirection {
+            type: null || 'walk' || 'bus',
+            routeNumber: null || Int,
+        }
+        */
+
+        const routeSummary = [];
+
+        let walkOnlyRoute = true;
+        for (let i = 0; i < directions.length; i++) {
+            walkOnlyRoute = walkOnlyRoute && (directions[i].type === 'walk');
+        }
+
+        // if walk only route, the first element of routeSummary will be 'walk'
+        if (walkOnlyRoute) {
+            routeSummary.push({
+                stopName: directions[0].name,
+                direction: { type: 'walk', routeNumber: null },
+                shouldStayOnBus: false,
+            });
+            routeSummary.push({
+                stopName: directions[directions.length - 1].name,
+                direction: { type: null, routeNumber: null },
+                shouldStayOnBus: false,
+            });
+        } else {
+            let indexOfFirstBus = -1;
+            for (let i = 0; i < directions.length; i++) {
+                if (directions[i].type === 'depart') {
+                    indexOfFirstBus = i;
+                    break;
+                }
+            }
+
+            for (let i = indexOfFirstBus; i < directions.length; i++) {
+                const newRouteSummary = { stopName: directions[i].name, shouldStayOnBus: false };
+                switch (directions[i].type) {
+                    case 'depart':
+                        newRouteSummary.direction = { type: 'bus', routeNumber: directions[i].routeNumber };
+                        break;
+                    case 'walk':
+                        newRouteSummary.direction = { type: 'walk', routeNumber: null };
+                        break;
+                    default:
+                        break;
+                }
+                routeSummary.push(newRouteSummary);
+            }
+
+            routeSummary.push({
```

Comment:
aren't u pushing the last direction twice (`directions[directions.length - 1]`)? b/c you iterate through the directions to push on `newRouteSummary` to `routeSummary` and that for-loop goes through the last direction: `for (let i = indexOfFirstBus; i < directions.length; i++)`, but then you're also do this `push` where you push `directions[directions.length - 1].name`. Unless this is supposed to be the case?